thirdeye/thirdeye-frontend/app/pods/components/rootcause-header/component.js

```diff
@@ -1,36 +1,104 @@
 import Component from '@ember/component';
-import { getProperties, computed } from '@ember/object';
+import { get, getProperties, computed } from '@ember/object';
+import { dateFormatFull } from 'thirdeye-frontend/helpers/utils';
 
 export default Component.extend({
   classNames: ['rootcause-header'],
-  sessionId: null, // 0
 
-  sessionName: null, // ""
+  /**
+   * Rootcause session identifier
+   * @type {Int}
+   */
+  sessionId: null,
+
+  /**
+   * Rootcause session title
+   * @type {string}
+   */
+  sessionName: null,
+
+  /**
+   * Rootcause session comments
+   * @Type {string}
+   */
+  sessionText: null,
+
+  /**
+   * Rootcause session dirty flag
+   * @type {boolean}
+   */
+  sessionModified: null,
+
+  /**
+   * Rootcause session last updated timestamp
+   * @type {Int}
+   */
+  sessionUpdatedTime: null,
 
-  sessionText: null, // ""
+  /**
+   * Rootcause session last updated author
+   * @type {string}
+   */
+  sessionUpdatedBy: null,
+
+  /**
+   * "Save" button callback
+   * @type {function}
+   */
+  onSave: null, // func()
+
+  /**
+   * "Copy" button callback
+   * @type {function}
+   */
+  onCopy: null, // func()
 
-  sessionModified: null, // true
+  /**
+   * Text edit callback
+   * @type {function}
+   */
+  onChange: null, // func(sessionName, sessionText)
 
-  onSave: null, // func(sessionName, sessionText)
+  /**
+   * Toggle for the comment textarea
+   * @type {boolean}
+   */
+  isCommentEditMode: computed.bool('sessionText'),
 
-  onCopy: null, // func(sessionName, sessionText)
+  /**
+   * Toggle for editing the session title
+   * @type {boolean}
+   */
+  isTitleEditMode: false,
 
   /**
-   * Toggles the comment textarea
+   * Formatted session last updated time
+   * @type {string}
    */
-  showComment: computed.bool('sessionText'),
+  sessionUpdatedTimeFormatted: computed('sessionUpdatedTime', function () {
+    return moment(get(this, 'sessionUpdatedTime')).format(dateFormatFull);
+  }),
 
   actions: {
+    /**
+     * "Save" button action
+     */
     onSave() {
       const { onSave } = getProperties(this, 'onSave');
```

Comment:
fyi, if `onSave` is a passed function. we can just to 
`this.attrs.onSave()`

Reply:
thx